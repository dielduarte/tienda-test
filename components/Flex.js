import styled from 'styled-components';

export const Row = styled.div`
  display: flex;
  align-items: center;
  flex-wrap: ${props => (props.wrap ? 'wrap' : 'nowrap')};
`;

export const Column = styled.div`
  display: flex;
  flex-direction: column;
`;
