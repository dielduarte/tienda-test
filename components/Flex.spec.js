import { shallow } from 'enzyme';
import 'jest-styled-components';

import { Row, Column } from './Flex';

describe('Flex', () => {
  it('Row', () => {
    const wrapper = shallow(<Row />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Row with wrap', () => {
    const wrapper = shallow(<Row wrap="true" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Column', () => {
    const wrapper = shallow(<Column />);
    expect(wrapper).toMatchSnapshot();
  });
});
