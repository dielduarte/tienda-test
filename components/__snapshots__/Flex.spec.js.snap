// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex Column 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
/>
`;

exports[`Flex Row 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

<div
  className="c0"
/>
`;

exports[`Flex Row with wrap 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<div
  className="c0"
  wrap="true"
/>
`;
